contributing/contributing.html

```diff
@@ -0,0 +1,51 @@
+<ul>
+<li>The requirements.txt file must be non-empty.</li>
+<li>All package requirements in the requirements.txt file <b>must</b> use the '>=' operator, never '=='. Security updates may require newer version of package dependencies to be installed in future builds.</li>
+<li>When versions are given as ">=", that should be interpreted as "last known working version".  The prime goal here is to ensure there is a record of what version of dependencies is known to produce the output the notebook author intended, ensuring the scientific results are reproducible.</li>
+</ul>
+
+</p>
+<h2>Authorship</h2>
+<p>A notebook must have assigned: <ol>
+<li>An author(s): the individual(s) who wrote the notebook</li>
+<li>An owner: the branch, team, or group that maintains the notebook content in the future. The owner is expected to periodically check the notebook for science relevance.</li></ol></p>
+<h2>PEP8 Compliance</h2>
+<p>In addition to the <a href="https://github.com/spacetelescope/style-guides/blob/master/guides/jupyter-notebooks.md">Notebook Style Guide</a>, any code in the notebook should be PEP8 compliant. PEP 8 compliance can be checked with the <a href="https://flake8.pycqa.org/en/latest/">flake8</a> and <a href="https://github.com/mattijn/pycodestyle_magic">pycodestyle_magic</a> packages.
+These packages can be installed used pip:</p>
+<p><code>$ pip install flake8 pycodestyle_magic</code></p>
+<p>Then add these two lines in a cell at the beginning of the notebook to test them:</p>
+<p><code>%load_ext pycodestyle_magic</code></p>
+<p><code>%flake8_on --ignore E261,E501,W291,W293</code></p>
+
+<h1>Step-by-step Guide</h1>
+This is a reproduction of the workflow instructions on the <a href="https://github.com/spacetelescope/style-guides/blob/master/guides/git-workflow.md">git workflow document</a>.
+<ol>
+<li>Read over the <a href="https://github.com/spacetelescope/style-guides/blob/master/guides/jupyter-notebooks.md">style guide</a>. The style parts are guidelines and it is not always necessary to follow them to the letter, but contributions are expected to generally follow this guide.</li>
+<li>Fork your target repository. See the preferred method of doing this on the <a href="https://github.com/spacetelescope/style-guides/blob/master/guides/git-workflow.md">workflow page</a>.</li>
+<li>Create a feature branch to add or modify content (<code>git checkout -b my-new-feature</code>).  You should aim to do one branch (and therefore one Pull Request) per notebook, as that will make review faster and easier. </li>
+<li>Create your notebook, starting with the <a href="https://github.com/spacetelescope/notebooks/tree/master/notebooks/MAST/templates/jupyter_template/jupyter_template.ipynb">the Jupyter Notebook template</a>. This should be organized into the <code>notebooks</code> directory or appropriate sub-directory, with each notebook in its <em>own</em> directory with no other notebooks.</li>
```

Comment:
Should the notebook template be somewhere general instead of specifically in MAST in the future?

Reply:
This is a good point. The MAST template is adapted from the one in the ["style guides" repo](https://github.com/spacetelescope/style-guides). Maybe the better solution is to submit a PR to the style guide people and link to that.